jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/AbstractDeploymentService.java

```diff
@@ -176,19 +179,40 @@ public void commonDeploy(DeploymentUnit unit, DeployedUnitImpl deployedUnit, Run
     
     @Override
     public void undeploy(DeploymentUnit unit) {
-        List<Integer> states = new ArrayList<Integer>();
-        states.add(ProcessInstance.STATE_ACTIVE);
-        states.add(ProcessInstance.STATE_PENDING);
-        states.add(ProcessInstance.STATE_SUSPENDED);
-        Collection<ProcessInstanceDesc> activeProcesses = runtimeDataService.getProcessInstancesByDeploymentId(unit.getIdentifier(), states, new QueryContext());
-        if (!activeProcesses.isEmpty()) {
-            throw new IllegalStateException("Undeploy forbidden - there are active processes instances for deployment " 
```

Comment:
So there will be no more exception when undeploying in production for example? I get it for "dev mode" but for production we probably should inform user that there are still active process instances.

Reply:
@MarianMacik yes that's correct. The kie-sever is the one that takes care of the mode (development or regular) and will call the right method depending on it. The jBPM runtime won't have any knowledge about the mode and will rely on the params received.

Reply:
@pefernan I discussed this with @mswiderski and we came to a conclusion that if `abortInstances` is false, then it should throw exception as it did until now if there are active processes. If `abortInstances` is true, then it should abort them and not throw exception.

Reply:
looks better now